src/ray/gcs/gcs_server/gcs_health_check_manager.cc

```diff
@@ -82,15 +82,39 @@ std::vector<NodeID> GcsHealthCheckManager::GetAllNodes() const {
   return nodes;
 }
 
+void GcsHealthCheckManager::MarkNodeHealthy(const NodeID &node_id) {
```

Comment:
this method is called from another thread (ray syncer's thread). do we need a mutex?

Reply:
The call comes from io context passed down to syner: https://github.com/ray-project/ray/blob/8ab5b2b38348734c90385cac72671861b5b1d67e/src/ray/common/ray_syncer/ray_syncer_bidi_reactor_base.h#L190-L192

Aren't  health check manager and syncer shared the same io context?

Reply:
OK, I found they're actually different io contexts:
- ray syncer: https://github.com/ray-project/ray/blob/8ab5b2b38348734c90385cac72671861b5b1d67e/src/ray/gcs/gcs_server/gcs_server.cc#L524-L525
- health check manager: https://github.com/ray-project/ray/blob/8ab5b2b38348734c90385cac72671861b5b1d67e/src/ray/gcs/gcs_server/gcs_server.cc#L291-L292

Reply:
Updated, I posted it to io context, which is the same implementation as `AddNode`.
https://github.com/ray-project/ray/blob/8ab5b2b38348734c90385cac72671861b5b1d67e/src/ray/gcs/gcs_server/gcs_health_check_manager.cc#L143-L153